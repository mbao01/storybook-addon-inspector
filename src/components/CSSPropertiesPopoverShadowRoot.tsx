"use client";

import root from "react-shadow";

import type { CSSPropertyPopoverProps } from "./types";
import { CSSPropertiesPopover } from "./CSSPropertiesPopover";
import styles from "../stylesheets/index.css?inline";

export const CSSPropertiesPopoverShadowRoot = ({
  id,
  tokens,
  computed,
  variables,
  open,
}: CSSPropertyPopoverProps) => {
  return (
    <root.div id={`${id}-shadow-root`} data-testid={`${id}-shadow-root`}>
      <CSSPropertiesPopover
        id={id}
        tokens={tokens}
        computed={computed}
        variables={variables}
        open={open}
      />
      <style type="text/css">{styles}</style>
    </root.div>
  );
};
