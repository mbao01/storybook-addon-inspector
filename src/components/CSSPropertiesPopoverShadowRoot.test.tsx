import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { CSSPropertiesPopoverShadowRoot } from "./CSSPropertiesPopoverShadowRoot";

describe("CSSPropertiesPopoverShadowRoot", () => {
  const mockProps = {
    id: "test-popover",
    tokens: [],
    computed: [],
    variables: [],
    open: true,
  };

  it("renders shadow root", () => {
    const { asFragment } = render(
      <CSSPropertiesPopoverShadowRoot {...mockProps} />,
    );

    expect(screen.getByTestId("test-popover-shadow-root")).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });
});
