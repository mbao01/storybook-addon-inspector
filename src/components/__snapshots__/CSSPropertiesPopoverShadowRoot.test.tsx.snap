// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CSSPropertiesPopoverShadowRoot > renders shadow root 1`] = `
<DocumentFragment>
  <div
    data-testid="test-popover-shadow-root"
    id="test-popover-shadow-root"
  />
</DocumentFragment>
`;
